book-content/chapters/01-setup-typescript.md

```diff
@@ -0,0 +1,143 @@
+
+Imagine you're building an IDE, and you want to give people warnings when they mis-type a function name or an object property. If you don't know the shapes of the variables, parameters and objects in your code, you'd have to resort to guesswork.
+
+But if you do know the types of everything in your app, you can begin implementing powerful IDE features like autocomplete, inline errors and automatic refactors.
+
+So, TypeScript aims to provide just enough strong typing to make working with JavaScript more pleasant and productive.
+
+## Tools for TypeScript Development
+
+Let's break down the tools you need in order to work with TypeScript:
+
+- An IDE: In order to write code, you need an editor or Integrated Development Environment. While you can use any IDE, the assumption in this book is that you are using Microsoft's Visual Studio Code. The TypeScript integration with VS Code is excellent, as you will see shortly. Install it from https://code.visualstudio.com if you haven't already.
+- An Execution Environment: You need somewhere to run your emitted JavaScript. This could be Node.js or a web browser like Chrome.
+- The TypeScript CLI: Node.js is needed in order to run the TypeScript CLI (command line interface). This tool converts your TypeScript to JavaScript, and warns you of any issues in your project.
+
+### Installing Node.js
+
+The Node.js installer can be downloaded from the [Node.js website](https://nodejs.org/).
+
+When you visit the site, you'll see two options: LTS and Current.
+
+LTS is short for "Long Term Support". This is the recommended version for production use. It's the most stable version, and the one we'll be using in this book.
+
+The Current version contains the latest features, but it's not recommended for production use.
+
+Click the LTS button to download the installer and follow the installation instructions.
+
+After running the installer, you can verify it installed correctly by opening a terminal and running the following command:
+
+```
+node -v
+```
+
+If Node.js is installed correctly, this command will display the version number. If you see an error message like "node command not found," it means the installation was not successful, and you should try again.
+
+### Alternative Package Management with PNPM
+
+Node.js includes the `npm` package manager by default.
+
+If you've worked with JavaScript repositories, you're likely familiar with `npm` and the `package.json` file. The `package.json` file represents all the packages we need to install to run the code in the repository.
+
+For example, in the repository for this material, we have a special CLI for running exercises along with helper packages and various other dependencies like `cross-fetch` and `nodemon`:
+
+```json
+// package.json
+
+{
+  "devDependencies": {
+    "@total-typescript/exercise-cli": "0.4.0",
+    "@total-typescript/helpers": "~0.0.1",
+    "cross-fetch": "~3.1.5",
```

Comment:
I guess this might make sense here but I came to prefer `undici` lately as it's the library that powers Node.js **native** fetch (see how it's included in the node's repo [here](https://github.com/nodejs/node/tree/10099bb3f7fd97bb9dd9667188426866b3098e07/deps/undici/src)). So it's guaranteed~ to be spec-compliant and future-proof.